bot/common/threads/points.py

```diff
@@ -57,42 +100,124 @@ async def send(self, message, user_id):
             date = datetime.now() - timedelta(days=int(days or "1"))
 
         contributions = await get_contributions(user_dao_id, date)
-        header = []
-        rows = []
-        for contribution in contributions:
-            fields = contribution.get("fields")
-            if not header:
-                header = [
-                    "Engagement",
-                    "status",
-                    "Date of Submission",
-                    "Date of Engagement",
-                    "Points",
-                ]
-            rows.append(
-                [
-                    fields.get("Activity"),
-                    fields.get("status"),
-                    fields.get("Date of Submission"),
-                    fields.get("Date of Engagement"),
-                    fields.get("Score"),
-                ]
-            )
+        # [0] is headers, [1] is a list of rows
+        contribution_rows = get_contribution_rows(contributions)
 
-        table = build_table(header, rows)
+        table = build_table(contribution_rows[0], contribution_rows[1])
         msg = f"```{table.draw()}```"
+        sent_message = None
         if not self.channel:
-            await channel.send(msg)
+            sent_message = await channel.send(msg)
+        # thread metadata returned from first interaction in a thread
+        # is not set if the first step is a trigger; this is a work-
+        # around for caching the contributions
```

Comment:
I think the issue may be that the metadata is set to None in the GetContributionCSVPrompt. Every step needs to pass the metadata if it is used

Reply:
Once it is passed all the way down we can remove the extra cache management logic in the other steps

Reply:
Is there a standard way for passing the metadata through the steps? or should I just retrieve it from the cache and return it from the step?

Reply:
Yes, I am pretty sure the standard way is to pass the metadata as the second return argument in a send all the way down the tree until you use it.